packages/telemetry/src/tooling-telemetry/data-processor.ts

```diff
@@ -262,29 +263,51 @@ function getInternalVsExternal(): InternalFeature {
 }
 
 /**
- * Read the manifest.json for the app and locate the tools id.
+ * Retrieves the source template configuration from either the standard manifest.json or
+ * the ui5.yaml based on the project type.
  *
- * @param appPath appPath Root folder path of Fiori app
- * @returns sourceTemplate section of data from manifest.json
+ * @param {string} appPath - The file system path to the application directory.
+ * @returns {Promise<SourceTemplate>} A promise that resolves to the source template configuration object.
  */
-async function getManifestSourceTemplate(appPath: string): Promise<SourceTemplate> {
-    let sourceTemplate: SourceTemplate = {};
+async function getSourceTemplate(appPath: string): Promise<SourceTemplate> {
+    const paths = {
+        manifest: path.join(appPath, 'webapp', 'manifest.json'),
+        appdescr: path.join(appPath, 'webapp', 'manifest.appdescr_variant'),
+        ui5Yaml: path.join(appPath, 'ui5.yaml')
```

Comment:
@nikmace Why support both `manifest.appdescr_variant` and `ui5.yaml` for adaptation projects?

Reply:
Here we need to determine what project type is being run. Adaptation Projects have `manifest.appdescr_variant` file and fiori apps have `manifest.json`. For Adaptation Projects we decided to put the toolsId inside `ui5.yaml` file. Or I need to prop drill and pass the project type this method.

Reply:
Thanks @nikmace for explanation.